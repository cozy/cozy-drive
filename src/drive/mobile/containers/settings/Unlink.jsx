import React from 'react'
import { withRouter } from 'react-router'
import { connect } from 'react-redux'
import { translate } from 'cozy-ui/react/I18n'
import SettingCategory, {
  ELEMENT_BUTTON
} from '../../components/SettingCategory'
import Modal from 'cozy-ui/react/Modal'
import {
  showUnlinkConfirmation,
  hideUnlinkConfirmation,
  unlink
} from '../../actions/unlink'
import { getClientSettings } from '../../../reducers/settings'

export const Unlink = (
  {
    t,
    showUnlinkConfirmation,
    hideUnlinkConfirmation,
    displayUnlinkConfirmation,
    unlink,
    clientSettings
  },
  { client }
) => (
  <div>
    <SettingCategory
      title={t('mobile.settings.unlink.title')}
      elements={[
        {
          type: ELEMENT_BUTTON,
          description: t('mobile.settings.unlink.description'),
          text: t('mobile.settings.unlink.button'),
          theme: 'danger-outline',
          onClick: showUnlinkConfirmation
        }
      ]}
    />
    {displayUnlinkConfirmation && (
      <Modal
        title={t('mobile.settings.unlink.confirmation.title')}
        description={t('mobile.settings.unlink.confirmation.description')}
        secondaryType="secondary"
        secondaryText={t('mobile.settings.unlink.confirmation.cancel')}
        secondaryAction={hideUnlinkConfirmation}
        primaryType="danger"
        primaryText={t('mobile.settings.unlink.confirmation.unlink')}
        primaryAction={() => unlink(client, clientSettings)}
      />
    )}
  </div>
)

const mapStateToProps = state => ({
  displayUnlinkConfirmation: state.mobile.ui.displayUnlinkConfirmation,
  clientSettings: getClientSettings(state)
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  showUnlinkConfirmation: () => dispatch(showUnlinkConfirmation()),
  hideUnlinkConfirmation: () => dispatch(hideUnlinkConfirmation()),
  unlink: (client, clientSettings) => {
    dispatch(unlink(client, clientSettings))
    ownProps.router.replace('/onboarding')
  }
})

export default withRouter(
  connect(mapStateToProps, mapDispatchToProps)(translate()(Unlink))
)
