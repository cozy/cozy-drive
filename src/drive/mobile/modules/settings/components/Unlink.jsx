import React, { Component } from 'react'
import PropTypes from 'prop-types'
import { withRouter } from 'react-router'
import { connect } from 'react-redux'
import { translate } from 'cozy-ui/react/I18n'
import SettingCategory, { ELEMENT_BUTTON } from './SettingCategory'
import {
  unlink,
  getClientSettings
} from 'drive/mobile/modules/authorization/duck'

export class Unlink extends Component {
  static contextTypes = {
    client: PropTypes.object.isRequired
  }
  render() {
    const { t, unlink, clientSettings } = this.props
    const { client } = this.context
    return (
      <div>
        <SettingCategory
          title={t('mobile.settings.unlink.title')}
          elements={[
            {
              type: ELEMENT_BUTTON,
              description: t('mobile.settings.unlink.description'),
              text: t('mobile.settings.unlink.button'),
              theme: 'danger-outline',
              onClick: () => unlink(client, clientSettings)
            }
          ]}
        />
      </div>
    )
  }
}
Unlink.propTypes = {
  t: PropTypes.func,
  unlink: PropTypes.func,
  clientSettings: PropTypes.object,
  router: PropTypes.object
}

const mapStateToProps = state => ({
  clientSettings: getClientSettings(state)
})

const mapDispatchToProps = (dispatch, ownProps) => ({
  unlink: (client, clientSettings) => {
    dispatch(unlink(client, clientSettings))
    ownProps.router.replace('/onboarding')
  }
})

export default withRouter(
  connect(
    mapStateToProps,
    mapDispatchToProps
  )(translate()(Unlink))
)
