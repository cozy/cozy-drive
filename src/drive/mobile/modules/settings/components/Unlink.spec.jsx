import React from 'react'
import { mount } from 'enzyme'
import { Unlink } from './Unlink'

jest.mock('cozy-ui/react/utils/color', () => ({
  getCssVariableValue: () => '#fff'
}))

describe('Unlink', () => {
  const client = jest.fn()
  const options = {
    context: {
      client
    }
  }
  const unlink = jest.fn()
  const t = jest.fn(s => s)
  const clientSettings = {
    data: 'foo'
  }
  const comp = mount(
    <Unlink t={t} unlink={unlink} clientSettings={clientSettings} />,
    options
  )

  afterEach(() => {
    jest.restoreAllMocks()
  })

  it('should render the Component', () => {
    expect(comp).toMatchSnapshot()
  })

  it('should call the unlink function', () => {
    const button = comp.find('Button')
    button.prop('onClick')()

    expect(unlink).toHaveBeenCalledWith(client, clientSettings)
  })
})
