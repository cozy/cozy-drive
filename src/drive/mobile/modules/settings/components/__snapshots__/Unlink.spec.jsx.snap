// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Unlink should render the Component 1`] = `
<Unlink
  clientSettings={
    Object {
      "data": "foo",
    }
  }
  t={
    [MockFunction] {
      "calls": Array [
        Array [
          "mobile.settings.unlink.title",
        ],
        Array [
          "mobile.settings.unlink.description",
        ],
        Array [
          "mobile.settings.unlink.button",
        ],
      ],
      "results": Array [
        Object {
          "isThrow": false,
          "value": "mobile.settings.unlink.title",
        },
        Object {
          "isThrow": false,
          "value": "mobile.settings.unlink.description",
        },
        Object {
          "isThrow": false,
          "value": "mobile.settings.unlink.button",
        },
      ],
    }
  }
  unlink={[MockFunction]}
>
  <div>
    <SettingCategory
      elements={
        Array [
          Object {
            "description": "mobile.settings.unlink.description",
            "onClick": [Function],
            "text": "mobile.settings.unlink.button",
            "theme": "danger-outline",
            "type": "ELEMENT_BUTTON",
          },
        ]
      }
      title="mobile.settings.unlink.title"
    >
      <div
        className="settings__category"
      >
        <h3
          className="settings__category-title"
        >
          mobile.settings.unlink.title
        </h3>
        <div
          key="0"
        >
          <p
            className="settings__subcategory__label"
          >
            mobile.settings.unlink.description
          </p>
          <Button
            className="settings__button"
            label="mobile.settings.unlink.button"
            onClick={[Function]}
            size="normal"
            tag="button"
            theme="danger-outline"
            type="submit"
          >
            <BaseButton
              className="settings__button"
              label="mobile.settings.unlink.button"
              onClick={[Function]}
              size="normal"
              tag="button"
              theme="danger-outline"
              type="submit"
            >
              <button
                className="c-btn c-btn--danger-outline settings__button"
                onClick={[Function]}
                title={null}
                type="submit"
              >
                <span>
                  <span
                    className={null}
                  >
                    mobile.settings.unlink.button
                  </span>
                </span>
              </button>
            </BaseButton>
          </Button>
        </div>
      </div>
    </SettingCategory>
  </div>
</Unlink>
`;
